File: src/main/scala/com/github/tachesimazzoca/akka/example/counter/Counter.scala
package com.github.tachesimazzoca.akka.example.counter

import akka.actor.Actor

protected class Counter extends Actor {
  def counter(n: Int): Receive = {
    case Counter.Increment => context.become(counter(n + 1))
    case Counter.Get => sender ! Counter.Count(n)
  }

  def receive = counter(0)
}

protected object Counter {
  case class Increment()
  case class Get()
  case class Count(n: Int)
}

File: java/akka-examples-counter/src/main/java/com/github/tachesimazzoca/akka/examples/counter/Counter.java
package com.github.tachesimazzoca.akka.examples.counter;

import akka.actor.UntypedActor;

public class Counter extends UntypedActor {
    private Integer counter = 0;

    public static enum Message {
        Reset, Increment, Get;
    }

    @Override
    public void onReceive(Object message) {
        if (message == Message.Reset) {
            counter = 0;
        } else if (message == Message.Increment) {
            counter++;
        } else if (message == Message.Get) {
            getSender().tell(counter, getSelf());
        } else {
            unhandled(message);
        }
    }
}
